src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -577,6 +577,24 @@ public static SyntaxNode GetInstance(this IInvocationOperation invocationOperati
             return invocationOperation.IsExtensionMethodAndHasNoInstance() ? invocationOperation.Arguments[0].Value.Syntax : invocationOperation.Instance.Syntax;
         }
 
+        public static ITypeSymbol GetInstanceType(this IOperation operation)
+        {
+            IOperation instance = operation switch
+            {
+                IInvocationOperation invocation => invocation.IsExtensionMethodAndHasNoInstance() ?
+                invocation.Arguments[0].Value :
+                invocation.Instance,
```

Comment:
Can we extract this into a helper, say `public static IOperation GetInstance(this IInvocationOperation invocationOperation)`, rename the existing `GetInstance` extension that returns a `SyntaxNode` to `GetInstanceSyntax`: 
```csharp
public static SyntaxNode GetInstanceSyntax(this IInvocationOperation invocationOperation) => invocationOperation.GetInstance().Syntax;

public static IOperation GetInstance(this IInvocationOperation invocationOperation) => invocationOperation.IsExtensionMethodAndHasNoInstance() ? invocationOperation.Arguments[0].Value : invocationOperation.Instance;

```